preprocess_era5_modellevels.py

```diff
@@ -0,0 +1,178 @@
+
+
+datelist = pd.date_range(
+    start=config["start_date"], end=config["end_date"], freq="d", inclusive="left"
+)
+
+for date in datelist[:1]:
+    print(date)
+
+    # Load data
+    u = load_modellevel_data("u", date)
+    v = load_modellevel_data("v", date)
+    q = load_modellevel_data("q", date)
+    sp = load_surface_data("sp", date) # in Pa
+    evap = load_surface_data("e", date)
+    cp = load_surface_data("cp", date)
+    lsp = load_surface_data("lsp", date)
+    tcw = load_surface_data("tcw", date) # kg/m2
+    tcrw = load_surface_data("tcrw", date) # kg/m2
+    precip = cp + lsp
+
+    # Get grid info
+    lat = u.latitude.values
+    lon = u.longitude.values
+    a_gridcell, l_ew_gridcell, l_mid_gridcell = get_grid_info(lat, lon)
+
+    # Calculate volumes
+    evap *= a_gridcell  # m3
+    precip *= a_gridcell  # m3
+
+    # Convert model levels to pressure values
+    filenamecsv = 'tableERA5model_to_pressure.csv'
+    df = pd.read_csv(os.path.join(config["input_folder"],filenamecsv))
+    a = df['a [Pa]'].to_xarray().rename(index='lev').sel(lev=modellevels)
+    b = df['b'].to_xarray().rename(index='lev').sel(lev=modellevels)
+    
+    sp_modellevels2 = sp.expand_dims({'lev':modellevels},axis=1)
+    p_modellevels = a + b*sp_modellevels2 # in Pa
+
+    #calculate the difference between the pressure levels
+    dp_modellevels = p_modellevels.diff(dim="lev") # in Pa
+
+    # Determine the fluxes and states
+    fa_e = u * q * dp_modellevels / g  # eastward atmospheric moisture flux
+    fa_n = v * q * dp_modellevels / g  # northward atmospheric moisture flux
+    cwv = q * dp_modellevels / g * a_gridcell[np.newaxis,np.newaxis,:] / density_water  # column water vapor (m3)
+
+    # Split in 2 layers 
+    boundary = 111 # set boundary model level 111
+    
+    # Integrate fluxes and states to upper and lower layer
+    lower_layer = dp_modellevels.lev > boundary 
+    upper_layer = ~lower_layer
+
+    # Integrate fluxes and state
+    fa_e_lower = fa_e.where(lower_layer).sum(dim="lev")
+    fa_n_lower = fa_n.where(lower_layer).sum(dim="lev")
+    w_lower = cwv.where(lower_layer).sum(dim="lev")
+
+    fa_e_upper = fa_e.where(upper_layer).sum(dim="lev")
+    fa_n_upper = fa_n.where(upper_layer).sum(dim="lev")
+    w_upper = cwv.where(upper_layer).sum(dim="lev")
+   
+    print(
+        "Check calculation water vapor over two layers, this value should be zero:",
+        (cwv.sum(dim="lev") - (w_upper + w_lower)).sum().values,
+    )
+
+    tcwm3 = tcw* a_gridcell[np.newaxis,:] / density_water # m3
+    
+    print(
+        "Check calculation water vapor with column water vapour, this value should be close to zero:",
+        (cwv.sum(dim="lev") - (tcwm3)).sum().values,
+    )
+    # is niet hetzelfde.. hoe erg is dat?
+    
+    # Change units to m3
+    # TODO: Check this! Change units before interp is tricky, if not wrong
+    total_seconds = config["timestep"] / config["divt"]
```

Comment:
timestep and diet have been deleted from the config file, there is now 1 config variable called tartet_freq (see preprocessing script era5 for pressure levels): 

Suggested code:
    # Change units to m3, based on target frequency (not incoming frequency!)
    target_freq = config['target_frequency']
    total_seconds = pd.Timedelta(target_freq).total_seconds()